osu.Game/Beatmaps/Drawables/Cards/BeatmapCardContentBackground.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Overlays;
+
+namespace osu.Game.Beatmaps.Drawables.Cards
+{
+    public class BeatmapCardContentBackground : ModelBackedDrawable<IBeatmapSetOnlineInfo>
+    {
+        public IBeatmapSetOnlineInfo BeatmapSet
+        {
+            get => Model;
+            set => Model = value;
+        }
+
+        public new bool Masking
+        {
+            get => base.Masking;
+            set => base.Masking = value;
+        }
+
+        public BindableBool Dimmed { get; private set; } = new BindableBool();
+
+        protected override double LoadDelay => 500;
+
+        protected override double TransformDuration => 400;
+
+        [BackgroundDependencyLoader]
+        private void load(OverlayColourProvider colourProvider)
+        {
+            InternalChild = new Box
+            {
+                RelativeSizeAxes = Axes.Both,
+                Colour = colourProvider.Background2
+            };
+        }
+
+        protected override DelayedLoadWrapper CreateDelayedLoadWrapper(Func<Drawable> createContentFunc, double timeBeforeLoad)
+            => new DelayedLoadUnloadWrapper(createContentFunc, timeBeforeLoad);
+
+        protected override Drawable? CreateDrawable(IBeatmapSetOnlineInfo? model)
+        {
+            if (model == null)
+                return null;
+
+            return new BufferedBackground(model)
+            {
+                RelativeSizeAxes = Axes.Both,
+                Dimmed = { BindTarget = Dimmed }
+            };
+        }
+
+        private class BufferedBackground : BufferedContainer
```

Comment:
I'm very wary of the performance of this, especially given how many of these are going to be on the screen at once. I can't see an overhead at my end, but this would likely be the largest use of uncached `BufferedContainer`s to date.

In addition, I can't seem to see the artifacts that you reported - is this supposed to be visible in the tests? I've tried all kinds of resolutions and it looks pretty clean here.

Reply:
The screenshot I took in the OP is from tests, yeah. It's definitely visible on my end, although much more so when hovering (makes everything darker, causing the cover to glow from the background in an ugly way). It may also be that on retina screens this may be less apparent? Not sure.

Reply:
I'll also note that messing with `MaskingSmoothness`, albeit hacky, also alleviates this somewhat (but then makes ugly edges).

Reply:
Very weird, I'm not on a retina screen either. What size window are you using to reproduce?

Wondering if there's some difference in shader handling in one of our drivers causing this..

Reply:
I basically see the masking artifact on any window size I try. But below are two undoctored fullscreen screenshots from my machine.

| 2560x1080 | 1920x1080 |
| :-: | :-: |
| ![2021-11-02-154504_2560x1080_scrot](https://user-images.githubusercontent.com/20418176/139870450-047ab3d8-371a-4a1a-aff9-edcaa40c3cb0.png) | ![2021-11-02-154516_1920x1080_scrot](https://user-images.githubusercontent.com/20418176/139870471-e7e97a56-770c-49b7-a08d-068384f2405f.png) |

In the interest of moving forward with this, though, I'll try alternative solutions that are most likely going to boil down to shrinking the cover image by 1 unit in all directions. I recall playlists items do similar. I tried this already once but couldn't get it perfect, but I'll try again.

Reply:
Have rewritten this entire thing to be shorter and use DLUW. Even managed to drop the buffered container thanks to your discovery that simply changing it out to a composite drawable fixes the ugly border (and it does look to fix it).